support-workers/src/main/scala/com/gu/support/workers/lambdas/CreatePaymentMethod.scala

```diff
@@ -65,6 +65,7 @@ class CreatePaymentMethod(servicesProvider: ServiceProvider = ServiceProvider)
       state.user,
       state.giftRecipient,
       state.product,
+      PaymentProvider.fromPaymentMethod(paymentMethod.left.toOption),
```

Comment:
when serialising the state, it just looks up the correct PaymentProvider from the payment method.

Other approaches could be to pass it through like we do for everything else `state.paymentProvider` or we could even `map` the encoder to add it to the JSON if we want to be clever (the latter would lose the compile safety but we could still unit test it)